plugins/pipe_text.py

```diff
@@ -0,0 +1,59 @@
+import sublime
+import sublime_plugin
+from subprocess import run, PIPE
+import time
+from datetime import datetime
+
+
+class PipeTextCommand(sublime_plugin.TextCommand):
+    """Pipe text from ST - the selections, if any, otherwise the entire buffer contents
+       - to the specified shell command.
+       Useful for formatting XML or JSON in a quick and easy manner.
+       i.e. a workaround for https://github.com/sublimehq/sublime_text/issues/3294
+    """
+    def run(self, edit, shell_command):
+        if all((region.empty() for region in self.view.sel())):
```

Comment:
Redundant parentheses.

Also, this condition allows empty regions, which I imagine would function in a weird way below. Did you want `any(not)` (or `not all`)? Or use the region's line if it is empty?

Since regions have `__len__` implemented, and that is also referenced in the default `__bool__` implementation, you can reduce this to `if not all(self.view.sel()):`.